src/connect.rs

```diff
@@ -797,13 +799,17 @@ mod socks {
 
     pub(super) async fn connect(
         proxy: ProxyScheme,
-        dst: Destination,
+        dst: Uri,
         dns: DnsResolve,
     ) -> Result<super::Conn, BoxError> {
         let https = dst.scheme() == Some(&Scheme::HTTPS);
-        let original_host = dst.host().to_owned();
+        let original_host = dst.host().unwrap().to_owned();
```

Comment:
This is a bit more panicky than it used to be, as `Destination` always had a `host()`. We could simply return an error here. Or how would you handle this?

Reply:
I used an `ok_or` now as used in other places in the file. Hope that's fine.